doc/api/stream.md

```diff
@@ -1359,6 +1364,19 @@ The `writable._writev()` method is prefixed with an underscore because it is
 internal to the class that defines it, and should never be called directly by
 user programs.
 
+#### writable.\_final(callback)
+
+* `callback` {Function} Call this function (optionally with an error
+  argument) when you are done writing any remaining data.
+
+Note: **This function MUST NOT be called directly.**  It MAY be implemented
+by child classes, and if so, will be called by the internal Writable
+class methods only.
+
+When the stream ends this function will be called before the stream closes,
+useful if you need to close a resource or write some data that you had buffered.
+This function is completely optional to implement.
```

Comment:
> I think we should mention in the _flush documentation how the two relate

Mentioning this with flush asks the question, why would one use flush instead of this, and I'd probably recommend using this instead of flush but we had decided to postpone the discussion on whether to soft depreciate flush or not.